internal/app/tfsec/rules/aws046.go

```diff
@@ -71,19 +67,18 @@ func init() {
 
 			if statementBlocks := block.GetBlocks("statement"); statementBlocks != nil {
 				for _, statementBlock := range statementBlocks {
-					if effect := statementBlock.GetAttribute("effect"); effect != nil {
-						if effect.Type() == cty.String && strings.ToLower(effect.Value().AsString()) == "deny" {
-							continue
-						}
+					if hasEffectDeny(statementBlock) {
```

Comment:
I appreciate it goes against the principle of DRY, but we want to keep the checks atomic so they can be deprecated with ease so trying to avoid functions spanning multiple checks.

As it is, this can be achieved with 

```
if statementBlock.HasChild("effect") && statementBlock.GetAttribute("effect").Equals("deny", block.IgnoreCase) {
	continue
}
```

Reply:
Oh I can totally get behind this! 👍